tests/regression/test_issue_1608.cpp

```diff
@@ -0,0 +1,63 @@
+// https://github.com/pocl/pocl/issues/1608
+// triggers on some LLVM versions
+// "LLVM ERROR: Instruction Combining did not reach a fixpoint after 1 iterations"
+
+#define CL_HPP_ENABLE_EXCEPTIONS
+#define CL_HPP_MINIMUM_OPENCL_VERSION 120
+#define CL_HPP_TARGET_OPENCL_VERSION 120
+#include <CL/opencl.hpp>
+#include <iostream>
+
+const char *SOURCE = R"RAW(
+float fn1(float *b, int c, int d) {
+  float e;
+  if (c * d)
+    e = b[0];
+  return e;
+}
+void fn2(int g, int h) {
+  float B[8], C[8], *p = 0;
+  int i = get_local_id(0);
+  for (int k = 0; k < g; ++k) {
+    int l = get_local_id(0) * get_local_id(1) * i * i * i;
+    p[l / 8 + l % 8] = B[0];
+  }
+  float f = fn1(0, i / h, i % h);
+  for (int m = 0; m < 8; ++m)
+    for (int n = 0; n < 2; ++n)
+      C[m] = mad(B[m], f, C[m]);
+}
+__kernel void krnl(int g) {
+  fn2(g, 1);
+}
+)RAW";
+
+int main(int argc, char *argv[]) {
+  int err = EXIT_SUCCESS;
```

Comment:
`cl_int`for portability (`cl_int` might not be equivalent to `int`) and some OpenCL error for avoiding false positives.
```suggestion
  cl_int err = CL_INVALID_VALUE;
```

Reply:
there is `int err = program.getInfo<>(CL_PROGRAM_BINARIES, &binaries);` in a few tests that you might want to fix, too